### PR TITLE
Serde Serializers for Record and Value objects

### PR DESCRIPTION
**Pull request by jonas32**
This PR adds the widely used Serializer for [Serde](https://github.com/serde-rs/serde) to the Value, Record, Bins, Key and BatchRead structs. (Serializer in value.rs made by @j-brn)
Its useful when using the Aerospike Client for REST APIs returning JSON with for example actix and does not affect the performance in any way if not used.
Without, you have to build own structs around the Value object and fill them with big match blocks.
Many Database drivers include it for that reason (for example Postgres and MongoDB).

### CODE REVIEW

**Review by jhecking — COMMENTED** [review]
I think this is a good idea and the implementation looks good to me. (Minor comments below.)

Even though there is no runtime overhead if the serialization functionality is not being used, you are still adding a new dependency on the `serde` crate, which might not be needed by most users. Maybe we should add a `serde` feature to the `aerospike` package to make this feature conditional?

File: tests/src/kv.rs [hunk 1]
```diff
@@ -16,9 +16,9 @@ use aerospike::operations;
 use aerospike::{
     as_bin, as_blob, as_geo, as_key, as_list, as_map, as_val, Bins, ReadPolicy, Value, WritePolicy,
 };
-use std::collections::HashMap;
-
 use env_logger;
+use serde_json::json;
```
[the remark this answers is not in the document]
Comment (jonas32) [reply tests/src/kv.rs]: Done. In addition, i fixed an import caused by as_map.

File: src/value.rs [hunk 2]
```diff
@@ -733,3 +736,47 @@ mod tests {
         assert_eq!(as_geo!(string), as_geo!(str));
     }
 }
+
+impl Serialize for Value {
```
[the remark this answers is not in the document]
Comment (jonas32) [reply src/value.rs:714]: I added tests for map and list serialization. This contains most possible datatypes.

**Comment by jonas32** [comment]
Ok, i feature gated the whole serialization content into the "serialization" feature.
You will need `cargo build --features "serialization"` or `cargo test --features "serialization"` to include it.
Probably we should adjust CI to build once without and once with the feature.
I didnt do it yet because i saw some messenger hooks on it and didnt want to spam the whole team when testing.

**Comment by jhecking** [comment]
> Ok, i feature gated the whole serialization content into the "serialization" feature.
> You will need `cargo build --features "serialization"` or `cargo test --features "serialization"` to include it.

Awesome! Thanks.

> Probably we should adjust CI to build once without and once with the feature.
> I didnt do it yet because i saw some messenger hooks on it and didnt want to spam the whole team when testing.

Agree. And you can go ahead and update `.travis.yaml` to add the additional testing steps.

**Comment by jonas32** [comment]
Would you like to run the feature as an independent build job or just run the `test --features` command right after the normal `test` command in the same script section?

**Comment by jhecking** [comment]
Good question. I think it will probably be faster to run both tests in the same Travis job, so let's do that.

**Comment by jonas32** [comment]
I updated both AppVeyor and Travis. 
AppVeyor only runs the cargo test for both and Travis runs build, test and README.md test for both.
Running all 3 steps twice in Travis might be redundant, but i think its the safer way (see the failed Travis build before at 59987c9)

**Review by jhecking — APPROVED** [review]
Looks good to me!

Thanks for another great contribution to the client! 🙏

**Comment by jhecking** [comment]
I'm going to go ahead and merge this, unless you were planning on making any further changes?

**Comment by jonas32** [comment]
You can go on. This features is finished.

**Pull request merged**